test/notificationSpec.js

```diff
@@ -358,62 +372,241 @@ describe('Notifications', function () {
       sinon.assert.calledWithMatch(listener.secondCall, {
-        type: 'transfer',
-        resource: transferPrepared
+        jsonrpc: '2.0',
+        id: null,
+        method: 'notify',
+        params: {
+          event: 'transfer.update',
+          resource: transfer
+        }
+      })
+    })
+
+    it('gets a 400 when subscribing without an id', function * () {
+      const listener = sinon.spy()
+      this.socket.on('message', (msg) => listener(JSON.parse(msg)))
+
+      this.socket.send(JSON.stringify({
+        jsonrpc: '2.0',
+        id: null,
+        method: 'subscribe_account',
+        params: { eventType: '*', accounts: ['http://localhost/accounts/alice'] }
+      }))
+
+      // TODO: Is there a more elegant way?
+      yield timingHelper.sleep(50)
+      assert.equal(listener.called, false)
+    })
+
+    it('gets a 400 when subscribing to an invalid account', function * () {
+      const listener = sinon.spy()
+      this.socket.on('message', (msg) => listener(JSON.parse(msg)))
+
+      this.socket.send(JSON.stringify({
+        jsonrpc: '2.0',
+        id: 2,
+        method: 'subscribe_account',
+        params: { eventType: '*', accounts: ['foo'] }
+      }))
+
+      // TODO: Is there a more elegant way?
+      yield timingHelper.sleep(50)
+
+      sinon.assert.calledOnce(listener)
+      sinon.assert.calledWithMatch(listener.firstCall, {
+        jsonrpc: '2.0',
+        id: 2,
+        error: { code: 400, message: 'Invalid account: foo' }
+      })
+    })
+
+    it('gets a 400 when subscribing with invalid parameters', function * () {
+      const listener = sinon.spy()
+      this.socket.on('message', (msg) => listener(JSON.parse(msg)))
+
+      this.socket.send(JSON.stringify({
+        jsonrpc: '2.0',
+        id: 2,
+        method: 'subscribe_account',
+        params: { eventType: '*', accounts: 'foo' }
+      }))
+
+      // TODO: Is there a more elegant way?
+      yield timingHelper.sleep(50)
+
+      sinon.assert.calledOnce(listener)
+      sinon.assert.calledWithMatch(listener.firstCall, {
+        jsonrpc: '2.0',
+        id: 2,
+        error: { code: 400, message: 'Invalid params' }
       })
-      sinon.assert.calledWithMatch(listener.thirdCall, {
-        type: 'transfer',
-        resource: transferCancelled
+    })
+
+    it('gets a 403 when subscribing to an account without permission', function * () {
+      const listener = sinon.spy()
+      this.socket.on('message', (msg) => listener(JSON.parse(msg)))
+
+      this.socket.send(JSON.stringify({
+        jsonrpc: '2.0',
+        id: 2,
+        method: 'subscribe_account',
+        params: { eventType: '*', accounts: ['http://localhost/accounts/bob'] }
+      }))
+
+      // TODO: Is there a more elegant way?
+      yield timingHelper.sleep(50)
+
+      sinon.assert.calledOnce(listener)
+      sinon.assert.calledWithMatch(listener.firstCall, {
+        jsonrpc: '2.0',
+        id: 2,
+        error: { code: 403, message: 'Not authorized' }
```

Comment:
I think it would be good if these errors also included the `id` field we send in the body for normal HTTP responses. The HTTP code is too general and the message isn't machine-readable